docs/source/nlp/punctuation_and_capitalization.rst

```diff
@@ -0,0 +1,345 @@
+| training_ds.shuffle                       | bool            | Whether to shuffle the training data                                                                         |
++-------------------------------------------+-----------------+--------------------------------------------------------------------------------------------------------------+
+| training_ds.num_samples                   | integer         | Number of samples to use from the training dataset, -1 - to use all                                          |
++-------------------------------------------+-----------------+--------------------------------------------------------------------------------------------------------------+
+| training_ds.batch_size                    | integer         | Training data batch size                                                                                     |
++-------------------------------------------+-----------------+--------------------------------------------------------------------------------------------------------------+
+| validation_ds.text_file                   | string          | Name of the text file for evaluation, located at `data_dir`                                                  |
++-------------------------------------------+-----------------+--------------------------------------------------------------------------------------------------------------+
+| validation_ds.labels_file                 | string          | Name of the labels dev file located at `data_dir`, such as `labels_dev.txt`                                  |
++-------------------------------------------+-----------------+--------------------------------------------------------------------------------------------------------------+
+| validation_ds.shuffle                     | bool            | Whether to shuffle the dev data                                                                              |
++-------------------------------------------+-----------------+--------------------------------------------------------------------------------------------------------------+
+| validation_ds.num_samples                 | integer         | Number of samples to use from the dev set, -1 mean all                                                       |
++-------------------------------------------+-----------------+--------------------------------------------------------------------------------------------------------------+
+| validation_ds.batch_size                  | integer         | Dev set batch size                                                                                           |
++-------------------------------------------+-----------------+--------------------------------------------------------------------------------------------------------------+
+| optim.name                                | string          | Optimizer to use for training                                                                                |
```

Comment:
@ericharper optimizer needs its own page in the core doc with just a reference here. Better not to have have duplicatations.